replicate/common/strings.go

```diff
@@ -70,3 +70,27 @@ func StringToPatternList(list string) (result []*regexp.Regexp) {
 
 	return
 }
+
+// GetDifferenceBetweenStringLists returns any entry that is in list a, but not in list b
+func GetDifferenceBetweenStringLists(a, b []string) []string {
+	if len(b) == 0 {
+		return a
+	}
+
+	if len(a) == 0 {
+		return []string{}
+	}
```

Comment:
Shorter: In the second case, you can _also_ just `return a`:

```suggestion
	if len(b) == 0 || len(a) == 0 {
		return a
	}
```